crates/biome_js_analyze/src/analyzers/nursery/no_misrefactored_shorthand_assign.rs

```diff
@@ -0,0 +1,155 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_syntax::{
+    AnyJsExpression, BinaryExpressionNodePosition, JsAssignmentExpression, JsAssignmentOperator,
+};
+use biome_rowan::{AstNode, BatchMutationExt};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Disallow shorthand assign when variable appears on both sides.
+    ///
+    /// This rule helps avoid potential bugs related to incorrect assignments or unintended
+    /// side effects that may occur during refactoring.
+    ///
+    /// Source: https://rust-lang.github.io/rust-clippy/master/#/misrefactored_assign_op
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// a += a + b
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// a -= a - b
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// a *= a * b
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// a += b
+    /// ```
+    ///
+    /// ```js
+    /// a = a + b
+    /// ```
+    ///
+    /// ```js
+    /// a = a - b
+    /// ```
+    ///
+    pub(crate) NoMisrefactoredShorthandAssign {
+        version: "next",
+        name: "noMisrefactoredShorthandAssign",
+        recommended: false,
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+pub struct RuleState {
+    replacement_expression: AnyJsExpression,
+}
+
+impl Rule for NoMisrefactoredShorthandAssign {
+    type Query = Ast<JsAssignmentExpression>;
+    type State = RuleState;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        if matches!(node.operator(), Ok(JsAssignmentOperator::Assign)) {
+            return None;
+        }
+
+        let right = node.right().ok()?;
+        let operator_token = node
+            .operator_token()
+            .ok()?
+            .kind()
+            .to_string()?
+            .split('=')
+            .nth(0)?;
```

Comment:
I think we can avoid the string allocation:
```suggestion
        let operator = node.operator_token().ok()?
        let operator = operator.text_trimmed();
        let operator = &operator[0..operator.len()-1];
```